.github/workflows/default.yaml

```
@@ -94,14 +94,26 @@ jobs:
     strategy:
       fail-fast: true
       matrix:
-        os: [ ubuntu-22.04, macos-14 ]
+        os:
+          - ubuntu-22.04
+          - macos-14
+        compiler:
+          - { CC: gcc, CXX: g++ }
+          - { CC: clang, CXX: clang++ }
         layer:
-          - layer-00-default
-          - layer-01-minimal
-          - layer-02-maximus
-          - layer-04-noauth-everything
+          - { layer: layer-00-default, nick: default }
+          - { layer: layer-01-minimal, nick: minimal }
+          - { layer: layer-02-maximus, nick: maximus }
+          - { layer: layer-04-noauth-everything, nick: noauth }
+        exclude:
+            # Non-clang testing on MacOS is too much work for very little gain
+            - { os: macos-14, compiler: { CC: gcc, CXX: g++ } }
 
     runs-on: ${{ matrix.os }}
+    name: build-tests(${{ matrix.os }}/${{ matrix.compiler.CC }}/${{ matrix.layer.nick }})
```

I would separate these three configuration sections a bit, especially since one of them is multiline.

```suggestion

    name: build-tests(${{ matrix.os }}/${{ matrix.compiler.CC }}/${{ matrix.layer.nick }})

```

I also think a comma is an overall better separator than a slash for this human-read info, but I do not insist on using a comma:

```suggestion

    name: build-tests(${{ matrix.os }},${{ matrix.compiler.CC }},${{ matrix.layer.nick }})

```